src/azure-cli/azure/cli/command_modules/keyvault/_params.py

```diff
@@ -75,6 +76,18 @@ def load_arguments(self, _):
         c.argument('no_self_perms', arg_type=get_three_state_flag(), help="Don't add permissions for the current user/service principal in the new vault.")
         c.argument('location', validator=get_default_location_from_resource_group)
 
+    with self.argument_context('keyvault recover') as c:
+        c.argument('vault_name', help='Name of the deleted vault', required=True, completer=None,
+                   validator=validate_deleted_vault_name)
+        c.argument('resource_group_name', resource_group_name_type, id_part=None, required=False,
```

Comment:
this is redundant, but maybe for the case that current principal doesn't have the permission to list soft deleted vaults?
either way, would be great to cross check, learn and make the CLI simpler and better

Reply:
We should keep `location` just in case "the current principal doesn't have the permission to list soft deleted vaults". For `resource_group_name`, we can keep it to align with `keyvault show`.